frontend/src/actionCreators/usersActionCreator.ts

```diff
@@ -66,20 +68,30 @@ export const fetchUserDetail = (id: API.IUserDetailParams) => (dispatch: Functio
     .finally(() => dispatch(hideLoading()));
 };
 
+const userToasts: LifecycleToasts = {
+  loadingToast: pimsToasts.USER_UPDATING,
+  successToast: pimsToasts.USER_UPDATED,
+  errorToast: pimsToasts.USER_ERROR,
+};
+
 export const getUpdateUserAction = (id: API.IUserDetailParams, updatedUser: any) => (
   dispatch: Function,
 ) => {
-  dispatch(request(reducerTypes.PUT_USER_DETAIL));
-  dispatch(showLoading());
-  return CustomAxios()
+  const axiosPromise = CustomAxios()
     .put(ENVIRONMENT.apiUrl + API.KEYCLOAK_USER_UPDATE(id), updatedUser)
     .then((response: AxiosResponse) => {
       dispatch(adminActions.updateUser(response.data));
-      dispatch(success(reducerTypes.PUT_USER_DETAIL));
-      dispatch(hideLoading());
-    })
-    .catch(() => dispatch(error(reducerTypes.PUT_USER_DETAIL)))
-    .finally(() => dispatch(hideLoading()));
+      return Promise.resolve(response);
+    });
+
+  return handleAxiosResponseWithToasts(
```

Comment:
I this this needs to be abstracted one more layer.  In this current implementation you are essentially wrapping CustomAxios to give it the toast feature.  But this means any place we want toasts we need to manually wrap.

Reply:
I think that is what we want isn't it? I could force all network requests to display toasts, but for that to work we would have to use extremely generic toast messages, and also we would end up displaying toasts for all async actions in the system, which would either result in a huge number of toasts, or toasts would end up frequently cancelling each other if we force unique ids on them.

Possibly I'm not understanding what you are recommending here. Not that handleAxiosResponseWithToasts is just an altered version of handleAxiosResponse. The original purpose of handleAxiosResponse was to provide all of the redux network actions against an axios response to prevent that code from having to be written for all async actions. The new method is similar, but shouldn't be applied to all network actions because we don't need toasts for every api call in the system.

Reply:
Couldn't you provide an abstraction of `CustomAxios` that includes toasts?  You could then pass in the toast messages you want to use.

Reply:
alright, I added the toasts to CustomAxios instead.

Reply:
Awesome, now it should be much easier to add the feature anywhere else we want it.